docs/protocol/content-node.md

```diff
@@ -0,0 +1,232 @@
+
+All registered content nodes are visible here: https://dashboard.audius.org/services/creator-node
+
+## Architecture
+
+**Web Server**
+
+The content node's core service is a web server with an HTTP API to process incoming requests and perform the following functions:
+
+- user & track metadata upload
+- user & track image upload
+- user track file upload
+- user & track data, metadata, and track file retrieval
+
+*The web server is a [NodeJS](https://nodejs.org) [Express app](https://expressjs.com/).*
+
+**Persistent Storage (Postgres)**
+
+It stores all data in a postgresql database and all images and metadata objects on its file system.
+
+Pointers to all content and metadata stored on disk are persisted in the Postgres DB.
+
+*Postgres is managed in the codebase using the [Sequelize ORM](https://sequelize.org/master/) which includes migrations, models and validations*
+
+**Redis**
+
+A [Redis client](https://redis.io/) is used for for resource locking, request rate limiting, and limited caching and key storage.
+
+*Redis is managed in the codebase through the [ioredis](https://github.com/luin/ioredis) npm package*
+
+**Track Segmenting**
+
+As defined by the [Audius protocol](https://whitepaper.audius.co), the content node uses [FFMPEG](https://ffmpeg.org/ffmpeg.html) to segment & transcode all uploaded track files before storing/serving.
+
+**Data Redundancy**
+
+As defined by the [Audius protocol](https://whitepaper.audius.co), all content is stored redundantly across multiple nodes to maximize availability. This is all done automatically - every node monitors every other node in the network to ensure minimum redundancy of all data, transfering files as required.
+
+## Configuration
+
+### Up and Running 
+
+#### Service Dependencies
+
+The Content Node depends on external services to run:
+* [Postgres](https://www.postgresql.org/)
+* [Redis](https://redis.io/)
+
+#### Docker Compose
+[Docker Compose](https://docs.docker.com/compose/) is used for a bit more automated management of the service dependencies.  
+**Make sure you have Docker and Docker-compose installed and operational.**
+
+To start the content node along with its services, run 
+```
+docker-compose -f docker-compose/docker-compose.yml up
+```
+
+*In development, it may be useful to use the `--build` flag to rebuild changes.*
+
+#### Docker
+The Audius Content Node is an evolving FOSS project. The application is dockerized for the ease of development and deployment of our user community. To use these images you will need to have Docker installed and operational.
+
+##### Environment Variables 
```

Comment:
we should just delete all of these. if we wanted to mention anything we should link to the config.js file

Reply:
love it, doing